components/storage/blockdevice/COMPONENT_FLASHIAP/FlashIAPBlockDevice.cpp

```diff
@@ -36,6 +36,8 @@ using namespace mbed;
 #define DEBUG_PRINTF(...)
 #endif
 
+#define BD_TYPE "FLASHIAP"
```

Comment:
Others may disagree, but I'm not a fan of this model, at all. Writing:

     const char * FlashIAPBlockDevice::get_type()
     {
         return BD_TYPE;
     }

is basically semantically meaningless. Extend that further and you can get

    REG1 = REG1_VALUE;
    REG2 = REG2_VALUE;
    <repeat for 100s of lines>

which I've seen in the past and it made me wince. It basically forces people to look in two places for a single piece of information. Pointless indirection.

Isn't this better?

     const char * FlashIAPBlockDevice::get_type()
     {
         return "FLASHIAP";
     }

No need to have a `#define` for a constant if that constant is used in exactly one place - doubly so if that place is the "specification" point itself.

Alternatively you could make the constants public in the drivers' header files so people `strcmp` against `FLASHIAP_TYPE` rather than `"FLASHIAP"`, meaning it is used in more than one place, but don't think that buys you anything.

Reply:
fixed